subprojects/core/src/integTest/groovy/org/gradle/api/tasks/NestedInputIntegrationTest.groovy

```diff
@@ -658,6 +654,86 @@ class NestedInputIntegrationTest extends AbstractIntegrationSpec {
         failure.assertHasCause("Cycles between nested beans are not allowed. Cycle detected between: 'nested' and 'nested.nested'.")
     }
 
+    def "duplicate names in nested iterable cause the build to fail"() {
+        buildFile << taskWithNestedInput()
+        buildFile << namedBeanClass()
+        buildFile << """
+            myTask.nested = [new NamedBean('name', 'value1'), new NamedBean('name', 'value2')]           
+        """
+
+        expect:
+        fails "myTask"
+        failure.assertHasDescription("Could not determine the dependencies of task ':myTask'.")
+        failure.assertHasCause("Nested iterables can only contain beans with unique names. Duplicate name: 'nested.name'.")
+    }
+
+    def "task with named nested beans is incremental"() {
+        buildFile << taskWithNestedInput()
+        buildFile << namedBeanClass()
+        buildFile << """                                   
+            myTask.nested = [new NamedBean(project.property('namedName'), 'value1'), new NamedBean('name', 'value2')]           
+        """
+        def taskPath = ':myTask'
+
+        when:
+        run taskPath, '-PnamedName=name1'
+        then:
+        executedAndNotSkipped taskPath
+
+        when:
+        run taskPath, '-PnamedName=name1'
+        then:
+        skipped taskPath
+
+        when:
+        run taskPath, '-PnamedName=different', '--info'
+        then:
+        executedAndNotSkipped taskPath
+        output.contains("Input property 'nested.different.class' has been added for task ':myTask'")
+        output.contains("Input property 'nested.name1.class' has been removed for task ':myTask'")
+    }
+
+    def "task with nested map is incremental"() {
```

Comment:
"incremental" in this context suggests to me that the task is executed in incremental mode. I don't know what a better name might be for these tests, though...

Reply:
I changed the names. Is that any better?

Reply:
👍